TypeScript error in `utils/i18n.ts`: Missing `fallbackType` property

### What version of `starlight` are you using?

0.26.1

### What version of `astro` are you using?

4.15.1

### What package manager are you using?

pnpm

### What operating system are you using?

Linux

### What browser are you using?

Chrome

### Describe the Bug

## Steps to reproduce

1. Create a new Starlight project
2. Run `npx tsc --noEmit`

## Error
```
../../node_modules/.pnpm/@astrojs+starlight@0.26.1_astro@4.15.1_@types+node@22.5.2_rollup@4.21.2_typescript@5.5.4_/node_modules/@astrojs/starlight/utils/i18n.ts:66:3 - error TS2322: Type '{ prefixDefaultLocale: boolean; redirectToDefaultLocale: false; }' is not assignable to type '"manual" | { prefixDefaultLocale: boolean; redirectToDefaultLocale: boolean; fallbackType: "redirect" | "rewrite"; }'.
  Property 'fallbackType' is missing in type '{ prefixDefaultLocale: boolean; redirectToDefaultLocale: false; }' but required in type '{ prefixDefaultLocale: boolean; redirectToDefaultLocale: boolean; fallbackType: "redirect" | "rewrite"; }'.

66   routing: {
     ~~~~~~~

  ../../node_modules/.pnpm/astro@4.15.1_@types+node@22.5.2_rollup@4.21.2_typescript@5.5.4/node_modules/astro/dist/core/config/schema.d.ts:877:13
    877             fallbackType: "redirect" | "rewrite";
                    ~~~~~~~~~~~~
    'fallbackType' is declared here.
  ../../node_modules/.pnpm/astro@4.15.1_@types+node@22.5.2_rollup@4.21.2_typescript@5.5.4/node_modules/astro/dist/core/config/schema.d.ts:874:9
    874         routing: "manual" | {
                ~~~~~~~
    The expected type comes from property 'routing' which is declared here on type '{ defaultLocale: string; locales: (string | { path: string; codes: [string, ...string[]]; })[]; routing: "manual" | { prefixDefaultLocale: boolean; redirectToDefaultLocale: boolean; fallbackType: "redirect" | "rewrite"; }; fallback?: Record<...> | undefined; domains?: Record<...> | undefined; }'


Found 1 error in ../../node_modules/.pnpm/@astrojs+starlight@0.26.1_astro@4.15.1_@types+node@22.5.2_rollup@4.21.2_typescript@5.5.4_/node_modules/@astrojs/starlight/utils/i18n.ts:66
```

## Cause

In the `devDependencies` for the `@astrojs/starlight` package, Astro version 4.10.2 is referenced. Sometime after that version, Astro added a new mandatory field to the `i18n.routing` config object called `fallbackType` which Starlight's `getAstroI18nConfig` function in `i18n.ts` does not provide.

## Solution

Update `packages/starlight/utils/i18n.ts#getAstroI18nConfig` to provide the missing field.

### Link to Minimal Reproducible Example

https://stackblitz.com/github/withastro/starlight/tree/main/examples/basics?file=README.md

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/fifty-gorillas-study.md
+++ b/.changeset/fifty-gorillas-study.md
@@ -1,0 +1,5 @@
+---
+'@astrojs/starlight': patch
+---
+
+Fixes type error when using Starlight with Astro v4.15

--- a/packages/starlight/__tests__/basics/i18n.test.ts
+++ b/packages/starlight/__tests__/basics/i18n.test.ts
@@ -303,7 +303,7 @@ function getAstroI18nTestConfig(i18nConfig: AstroUserConfig['i18n']): AstroConfi
 		...i18nConfig,
 		routing:
 			typeof i18nConfig?.routing !== 'string'
-				? { prefixDefaultLocale: false, ...i18nConfig?.routing }
+				? { prefixDefaultLocale: false, fallbackType: 'redirect', ...i18nConfig?.routing }
 				: i18nConfig.routing,
 	} as AstroConfig['i18n'];
 }

--- a/packages/starlight/utils/i18n.ts
+++ b/packages/starlight/utils/i18n.ts
@@ -70,6 +70,7 @@ function getAstroI18nConfig(config: StarlightConfig): NonNullable<AstroConfig['i
 				// Sites with a single non-root language different from the built-in default locale.
 				(!config.isMultilingual && config.locales !== undefined),
 			redirectToDefaultLocale: false,
+			fallbackType: 'redirect',
 		},
 	};
 }

--- a/packages/starlight/utils/i18n.ts
+++ b/packages/starlight/utils/i18n.ts
@@ -70,6 +70,8 @@ function getAstroI18nConfig(config: StarlightConfig): NonNullable<AstroConfig['i
 				// Sites with a single non-root language different from the built-in default locale.
 				(!config.isMultilingual && config.locales !== undefined),
 			redirectToDefaultLocale: false,
+			// TODO: remove this ignore comment for Astro v5
+			// @ts-ignore — Only used in Astro >=4.15.0, but Starlight supports ^4.8.6
 			fallbackType: 'redirect',
 		},
 	};

